src/store/index.ts

```diff
@@ -258,6 +263,15 @@ async function alertMatchedErrorAction(this: Store<any>, {
 
 /* --- MUTATIONS --- */
 
+function completeCodeWordMutation(state: any): void {
+    const codeParts = state.code.split(CODE_DELIMITER);
+    // Replace last (incomplete) word `codeWord`
+    codeParts.splice(codeParts.length - 1, 1, state.suggestedWord);
+
+    console.log(codeParts)
+    state.code = codeParts.join(CODE_DELIMITER);
+}
```

Comment:
Yes, looks like that. It seem to replace `code` with the suggested words. If suggestion is wrong then it ignores the user input. That looks wrong to me.